private_sharing/api_views.py

```diff
@@ -458,3 +470,102 @@ def post(self, request):
                 data_file.delete()
 
         return Response({"ids": ids}, status=status.HTTP_200_OK)
+
+
+class ProjectCreateAPIView(APIView):
+    """
+    Create a project via API
+
+    Accepts project name, description, and redirect_url as (required) inputs
+
+    The other required fields are auto-populated:
+    is_study:  set to False
+    leader:  set to member.name from oauth2 token
+    coordinator:  get from oauth2 token
+    is_academic_or_nonprofit: False
+    add_data:  false
+    explore_share:  false
+    short_description:  first 139 chars of long_description plus an ellipsis
+    active:  True
+    """
+
+    authentication_classes = (CustomOAuth2Authentication,)
+    permission_classes = (HasValidProjectToken,)
+
+    def get_short_description(self, long_description):
+        """
+        Return first 139 chars of long_description plus an elipse.
+        """
+        if len(long_description) > 140:
+            return "{0}…".format(long_description[0:139])
+        return long_description
+
+    def post(self, request):
+        """
+        Take incoming json and create a project from it
+        """
+        member = get_oauth2_member(request).member
+        serializer = ProjectAPISerializer(data=request.data)
+        if serializer.is_valid():
```

Comment:
Our other APIs use re-use the on site forms for validation: https://github.com/OpenHumans/open-humans/blob/master/open_humans/api_views.py

I think we should be doing the same here, to have consistency in validation (i.e. re-using `OAuth2DataRequestProjectForm`)